kernel/pci/src/lib.rs

```diff
@@ -129,7 +158,20 @@ pub struct PciBus {
 
 /// Scans all PCI Buses (brute force iteration) to enumerate PCI Devices on each bus.
 /// Initializes structures containing this information. 
-fn scan_pci() -> Vec<PciBus> {
+fn scan_pci() -> Result<Vec<PciBus>, &'static str> {
+    #[cfg(target_arch = "aarch64")] {
+        let mut config_space = PCI_CONFIG_SPACE.lock();
+        if config_space.is_none() {
+            let config = BOARD_CONFIG.pci_ecam;
+            let flags = memory::PteFlags::new().valid(true).writable(true);
+            let mapped = memory::map_frame_range(config.base_address, config.size_bytes, flags)?;
+            match mapped.into_borrowed_slice_mut(0, config.size_bytes) {
+                Ok(bsm) => config_space.insert(bsm),
+                Err((_, msg)) => return Err(msg),
+            };
+        }
+    }
```

Comment:
we're using `try_call_once()` for the PCI Bus, shouldn't we also use it for this? Is there any other reason to re-initialize or otherwise modify the `PCI_CONFIG_SPACE` variable more than once?

Reply:
We need to modify it, when using `pci_write_*`, but I will replace the option with Once.